docs/blocks/states.md

```diff
@@ -1,82 +1,98 @@
+    * Created by calling `BooleanProperty.create(String propertyName)`.
+  * `EnumProperty<E extends Enum<E>>`
+    * Implements `IProperty<E>`. Defines a property that can take on the values of an Enum class.
+    * Created by calling `EnumProperty.create(String propertyName, Class<E> enumClass)`.
+    * It is also possible to use only a subset of the Enum values (e.g. 4 out of 16 `DyeColor`s). See the overloads of `EnumProperty.create`.
+  * `DirectionProperty`
+    * This is a convenience implementation of `EnumProperty<Direction>`
+    * Several convenience predicates are also provided. For example, to get a property that represents the cardinal directions, call `DirectionProperty.create("<name>", Direction.Plane.HORIZONTAL)`; to get the X directions, `DirectionProperty.create("<name>", Direction.Axis.X)`
 
+Different blocks may share the same `IProperty<?>` object. Vanilla generally has separate ones for every single block, but it is merely personal preference.
 
-Using `BlockState`'s
----------------------
+!!! Note 
+    If your mod has an API or is meant to be interacted with from other mods, it is **highly** recommended that you instead place your `IProperty<?>`'s (and any classes used as values) in your API. That way, people can use your properties to interact with your blocks.
 
-`BlockState`, as we know now, is a powerful object. You can get the value of a property by calling `get(<PROPERTY>)`, passing it the `IProperty<>` you want to test.
-If you want to get an `BlockState` with a different set of values, simply call `with(<PROPERTY>, <NEW_VALUE>)` as mentioned above. This will return another of the pregenerated `BlockState`'s with the values you requested.
+After you've created your `IProperty<>` objects, override `Block#fillStateContainer(StateContainer.Builder)` in your Block class. In that method, call `StateContainer.Builder#add(...);`  with the parameters as every `IProperty<?>` you wish the block to have.
 
-You can get and put `BlockState`'s in the world using `setBlockState()` and `getBlockState()`.
+Every block will also have a "default" state that is automatically chosen for you. You can change this "default" state by calling the `Block#setDefaultState(BlockState)` method from your constructor. When your block is placed it will become this "default" state. An example from `DoorBlock`:
 
+```Java
+this.setDefaultState(
+    this.stateContainer.getBaseState()
+        .with(FACING, Direction.NORTH)
+        .with(OPEN, Boolean.valueOf(false))
+        .with(HINGE, DoorHingeSide.LEFT)
+        .with(POWERED, Boolean.valueOf(false))
+        .with(HALF, DoubleBlockHalf.LOWER)
+);
+```
+
+If you wish to change what `BlockState` is used when placing your block, you can overwrite `Block#getStateForPlacement(BlockItemUseContext)`. This can be used to -- for example -- set the direction of your block depending on where the player is standing when they place it.
```

Comment:
```suggestion
If you wish to change what `BlockState` is used when placing your block, you can overwrite `Block#getStateForPlacement(BlockItemUseContext)`. This can be used to, for example, set the direction of your block depending on where the player is standing when they place it.
```

or just "e.g."